invenio_rdm_records/resources/serializers/datacite/schema.py

```diff
@@ -538,14 +548,25 @@ def get_subjects(self, obj):
         for subject in subjects:
             entry = {"subject": subject.get("subject")}
 
-            id_ = subject.get("id")
-            if id_:
-                entry["subjectScheme"] = subject.get("scheme")
-                try:
-                    validator(id_)
-                    entry["valueURI"] = id_
-                except ValidationError:
-                    pass
+            subject_scheme = subject.get("scheme")
+            if subject_scheme:
+                entry["subjectScheme"] = subject_scheme
+
+            # Get identifiers and assign valueURI if scheme is 'url'
+            identifiers = subject.get("identifiers", [])
+            for identifier in identifiers:
+                if identifier.get("scheme") == "url":
+                    entry["valueURI"] = identifier.get("identifier")
+                    break
+
+            # Add id as valueURI if no 'url' scheme is found
+            if "valueURI" not in entry and subject.get("id"):
+                entry["valueURI"] = subject.get("id")
+
+            # Add props if it exists
+            props = subject.get("props", {})
+            if props:
+                entry["subjectProps"] = props
```

Comment:
minor: we can't add new fields, since they will probably be rejected by the DataCite REST API when registering metadata for the record's DOI.

---

Again, I assume DCAT XSLT makes use of this 🥲